scripts/workflow.py

```diff
@@ -0,0 +1,153 @@
+
+    def align(
+        self,
+        center_align: bool = False,
+        center_slice: Optional[Slice] = None,
+        pis: Optional[np.ndarray] = None,
+        overlap_fraction: Optional[float] = None,
+        max_iters: int = 1000,
+    ):
+        if center_align:
+            if overlap_fraction is not None:
+                logger.warning(
+                    "Ignoring overlap_fraction argument (unsupported in center_align mode)"
+                )
+            return self.center_align(center_slice, pis)
+        else:
+            assert overlap_fraction is not None, "overlap_fraction must be specified"
+            return self.pairwise_align(
+                overlap_fraction=overlap_fraction, pis=pis, max_iters=max_iters
+            )
+
+    def find_pis(self, overlap_fraction: float, max_iters: int = 1000):
+        pis = []
+        for i in range(len(self) - 1):
+            logger.info(f"Finding Pi for slices {i} and {i+1}")
+            pis.append(
+                pairwise_align(
+                    self.slices[i].adata,
+                    self.slices[i + 1].adata,
+                    s=overlap_fraction,
+                    numItermax=max_iters,
+                    maxIter=max_iters,
+                    verbose=True,
+                )
+            )
+        return pis
+
+    def pairwise_align(
+        self,
+        overlap_fraction: float,
+        pis: Optional[List[np.ndarray]] = None,
+        max_iters: int = 1000,
+    ):
+        if pis is None:
+            pis = self.find_pis(overlap_fraction=overlap_fraction, max_iters=max_iters)
+        new_slices = stack_slices_pairwise(self.slices_adata, pis)
+        return AlignmentDataset(slices=[Slice(adata=s) for s in new_slices])
+
+    def find_center_slice(
+        self, reference_slice: Optional[Slice] = None, pis: Optional[np.ndarray] = None
+    ) -> Tuple[Slice, List[np.ndarray]]:
+        if reference_slice is None:
+            reference_slice = self.slices[0]
+        center_slice, pis = center_align(
+            reference_slice.adata, self.slices_adata, pis_init=pis
+        )
+        return Slice(adata=center_slice), pis
+
+    def find_pis_init(
+        self, reference_slice: Optional[Slice] = None
+    ) -> List[np.ndarray]:
+        if reference_slice is None:
+            reference_slice = self.slices[0]
+        return [
+            match_spots_using_spatial_heuristic(reference_slice.adata.X, slice_.adata.X)
+            for slice_ in self.slices
+        ]
+
+    def center_align(
+        self,
+        reference_slice: Optional[Slice] = None,
+        pis: Optional[List[np.ndarray]] = None,
+    ):
+        if reference_slice is None:
+            reference_slice = self.slices[0]
+        if pis is None:
+            pis = self.find_pis_init(reference_slice)
+
+        _, new_slices = stack_slices_center(
+            center_slice=reference_slice.adata, slices=self.slices_adata, pis=pis
```

Comment:
This line passes `reference_slice.adata` as the `center_slice` to `stack_slices_center`. Shouldn't this be the newly configured center slice instead? Also it seems like the function `find_center_slice` isn't being used anywhere.